.github/workflows/continuous-deployment.yml

```diff
@@ -245,8 +245,15 @@ jobs:
           export CAMS_FEATURE_FLAG_CLIENT_ID="${{ secrets.LD_DEVELOPMENT_CLIENT_ID }}"
           npm run build --if-present
 
-      - name: Execute Test
-        run: CI=true npm test
+      - name: Execute Tests
+        if: ${{ vars.ENVIRONMENT != 'Main-Gov' && github.event_name != 'pull_request' }}
+        run: |
+          CI=true npm run test
+
+      - name: Execute Coverage
+        if: ${{ vars.ENVIRONMENT == 'Main-Gov' || (github.event_name == 'pull_request' && github.base_ref == 'main') }}
+        run: |
+          CI=true npm run coverage
```

Comment:
Do we need to have both of these jobs? `npm run coverage` runs all the tests to keep track of what is and is not covered. Failure output is similar.